airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/typing_deduping/JdbcTypingDedupingTest.kt

```diff
@@ -26,12 +31,20 @@ import org.jooq.impl.DSL
 abstract class JdbcTypingDedupingTest : BaseTypingDedupingTest() {
     protected var database: JdbcDatabase? = null
     private var dataSource: DataSource? = null
+    protected open val dialect: SQLDialect = SQLDialect.DEFAULT
+    protected val dslContext: DSLContext
+        get() = DSL.using(dialect)
+    protected open val nameTransformer: NamingConventionTransformer = StandardNameTransformer()
 
+    /**
+     * Get the config as declared in GSM (or directly from the testcontainer). This class will do
+     * further modification to the config to ensure test isolation.
+     *
+     * This getter MUST return a new instance on every invocation. Subclasses MAY modify the config
+     * object, so we want to prevent them from modifying the base config. The easiest way to achieve
+     * this is by returning `yourBaseConfig.deepCopy()`.
+     */
     protected abstract val baseConfig: ObjectNode
```

Comment:
is it really a getter if it creates a new instance at every call?

Reply:
thought about this more, and I think that if that instance creation is a requirement, this really shouldn't be a field with a getter, but rather an abstract function

Reply:
I think my concern is more around the fact that it's _necessary_. In a world where objects are immutable, passing a reference will obviously be faster, and most people just return references regardless

Reply:
swapped to `fun getBaseConfig`

Reply:
the mutability is really the killer here...